pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -269,15 +276,26 @@ func (e *EndpointController) processNextWorkItem() bool {
 	defer e.queue.Done(eKey)
 
 	err := e.syncService(eKey.(string))
+	e.handleErr(err, eKey)
+
+	return true
+}
+
+func (e *EndpointController) handleErr(err error, key interface{}) {
 	if err == nil {
-		e.queue.Forget(eKey)
-		return true
+		e.queue.Forget(key)
+		return
 	}
 
-	utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", eKey, err))
-	e.queue.AddRateLimited(eKey)
+	if e.queue.NumRequeues(key) < maxRetries {
+		glog.V(2).Infof("Error syncing endpoints for service %q: %v", key, err)
+		e.queue.AddRateLimited(key)
+		return
+	}
 
-	return true
+	glog.V(2).Infof("Dropping service %q out of the queue: %v", key, err)
```

Comment:
What I meant to say was this event of 'dropping a workqueue item after failing retries' is not seen as a Warning level log at some places (eg. https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/deployment_controller.go#L490).
But I agree that it should be warning. Changed it.